### PR TITLE
Added profile page , Dashboard and leaderboard functionalities

### PR DESCRIPTION
**Pull request by Rohanmond**

### CODE REVIEW

File: src/pages/Profile/Profile.css [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  align-items: center;
+  justify-content: center;
+}
+.profile-container {
+  box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
+  width: 95%;
+  min-height: 90%;
+  margin-bottom: 2rem;
+  border-radius: 0.5rem;
+  /* background-color: var(--blue-200); */
```
Comment (romabulani) [inline src/pages/Profile/Profile.css]: Remove comment if not required

File: src/pages/Profile/Profile.css [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+@media screen and (max-width: 500px) {
+  .profile-container {
+    width: 100%;
+    min-height: 100%;
+    box-shadow: none;
+    border: none;
+  }
+  .profile-nav {
+    justify-content: space-evenly;
+    /* font-size: 0.5rem; */
```
Comment (romabulani) [inline src/pages/Profile/Profile.css]: here also remove comment

File: src/pages/Profile/Profile.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  return (
+    <>
+      <NavBar />
+      <div className='profile-outer-container'>
+        <div className='profile-container'>
+          <div className='profile-nav'>
+            <NavLink to={'/profile/details'} className={toggleActive}>
+              Profile Information
+            </NavLink>
+            {/*address management and order management will be added later on*/}
```
Comment (romabulani) [inline src/pages/Profile/Profile.tsx]: Here also

**Review by PeeyushPrashant — APPROVED** [review]
good folder structure
nice to see that you have used TypeScript
and Firebase

**Review by romabulani — APPROVED** [review]
- Clean and modular code
- Functionality is working as expected
- Good to merge
- Suggestions: Remove the unwanted comments and empty file

File: src/pages/Profile/components/Dashboard/Dashboard.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import { collection, onSnapshot, query, where } from 'firebase/firestore';
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useAuth } from '../../../../contexts/auth-context';
+import { QuizModel } from '../../../../DataModel/quiz.model';
+import { db } from '../../../../firebase';
+import { UserType } from '../../../../types';
+import './Dashboard.css';
```
Comment (lomesshh) [inline src/pages/Profile/components/Dashboard/Dashboard.tsx:8]: suggestion: 
try to explore absolute paths it will improve readability

**Review by lomesshh — APPROVED** [review]
clean and modular code
got a glimpse of typescript from your code
looks good to merge

**Pull request merged**